### DISCUSSION TITLE
IPv6 addresses are not reachable from a container

### DISCUSSION THREAD

**Discussion by Zebradil** [k3s-io/k3s#9092] [Q&A] [answered]
k3s is installed with the following command:

```shell
curl -sfL https://get.k3s.io | \
    INSTALL_K3S_CHANNEL=latest \
    sh -s - \
        --disable-helm-controller \
        --disable=traefik \
        --disable=servicelb
```

in a simple `curl` container, created with `kubectl run curl -it --rm --image=quay.io/curl/curl --command -- /bin/sh`:

```
# ip route
default via 10.42.0.1 dev eth0
10.42.0.0/24 dev eth0 scope link  src 10.42.0.47
10.42.0.0/16 via 10.42.0.1 dev eth0

# ip -6 route
fe80::/64 dev eth0  metric 256
multicast ff00::/8 dev eth0  metric 256

# ping -c1 google.com
PING google.com (173.194.76.100): 56 data bytes
64 bytes from 173.194.76.100: seq=0 ttl=59 time=32.569 ms

--- google.com ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 32.569/32.569/32.569 ms

# ping -c1 ipv6.google.com
PING ipv6.google.com (2a00:1450:400c:c09::66): 56 data bytes
ping: sendto: Network unreachable
```

I tried to add a default route with `ip -6 route add default via fe80:: dev eth0`. This makes `Network unreachable` to disappear, but now `ping` just hangs.

Where should I look next?

**Comment by brandond** [comment, marked as answer]
> Where should I look next?

The docs?
https://docs.k3s.io/installation/network-options#dual-stack-ipv4--ipv6-networking

By default the cluster is single-stack IPv4.

**Reply by Zebradil** [reply]
Thanks. For some reason I didn't realize that no ipv6 on pods also means no ipv6 egress.
